pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,386 @@
+            ...
+            >>> # inversion
+            >>> Inversion = pf.dsp.RegularizedSpectrumInversion.from_frequency_range(
+            >>>     sweep, [50, 16e3], beta='max')
+            >>> inverted = Inversion.invert()
+            ...
+            >>> # obtain the scaled regularization function
+            >>> regularization = Inversion.regularization * Inversion.beta
+            ...
+            >>> pf.plot.use()
+            >>> fig, axes = plt.subplots(2,1)
+            >>> pf.plot.freq(sweep, ax=axes[0], label='sweep')
+            >>> pf.plot.freq(regularization, ax=axes[0], label='regularization')
+            >>> axes[0].axvline(50, color='k', linestyle='--',
+            ...     label='frequency range')
+            >>> axes[0].axvline(16e3, color='k', linestyle='--')
+            >>> axes[0].legend(loc='lower center')
+            ...
+            >>> pf.plot.freq(inverted, ax=axes[1], color='p',
+            ...     label='inverted with regulariztion')
+            >>> pf.plot.freq(1 / (sweep+1e-10), ax=axes[1], color='y',
+            ...     linestyle=':', label='inverted without regulariztion')
+            >>> axes[1].axvline(50, color='k', linestyle='--',
+            ...     label='frequency range')
+            >>> axes[1].axvline(16e3, color='k', linestyle='--')
+            >>> axes[1].set_ylim(-120, -20)
+            >>> axes[1].legend(loc='lower center')
+
+        Invert a headphone transfer function (HpTF), regularize the inversion at
+        high frequencies, and use a band-pass as target function. Note that the
+        equalized HpTF, which is obtained from a convolution of the HpTF with its
+        inverse filter, approaches the target function in both, time and frequency.
+
+        .. plot::
+
+            >>> import pyfar as pf
+            >>> import numpy as np
+            >>>
+            >>> # Hedphone transfer function to be inverted
+            >>> hptf = pf.signals.files.headphone_impulse_responses()[0, 0].flatten()
+            >>> # Regularize inversion at high frequencies
+            >>> regularization = pf.dsp.filter.low_shelf(
+            ...     pf.signals.impulse(hptf.n_samples), 4e3, -20, 2, 'II')
+            >>> # Use band-pass as target function
+            >>> target = pf.dsp.filter.butterworth(
+            ...     pf.signals.impulse(hptf.n_samples), 4, [100, 16e3], 'bandpass')
+            >>> target = pf.dsp.time_shift(target, 125, 'cyclic')
+            >>>
+            >>> # Regulated inversion
+            >>> Inversion = pf.dsp.RegularizedSpectrumInversion.from_magnitude_spectrum(
+            ...     hptf, regularization, beta=.1, target=target)
+            >>> inverted = Inversion.invert()
+            >>>
+            >>> # Plot the result
+            >>> ax = pf.plot.time_freq(hptf, label='HpTF')
+            >>> pf.plot.time_freq(inverted, label='HpTF inverted')
+            >>> pf.plot.time_freq(hptf * inverted, label='Equalized HpTF')
```

Comment:
```suggestion
            >>>
            >>> # Plot the result
            >>> ax = pf.plot.time_freq(hptf, label='HpTF')
            >>> pf.plot.time_freq(inverted, label='HpTF inverted')
            >>> pf.plot.time_freq(hptf * Inversion, label='Equalized HpTF')
```

so that the inversion class returns the invert for multiplications
would it be cool to do sth like this?

Reply:
It would be more convenient but less implicit. I would prever the current way. However, `invert` is not a function anymore, but a property. So the call simplifies to `Inversion.invert`.